group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinator.java

```diff
@@ -215,4 +223,86 @@ CompletableFuture<OffsetDeleteResponseData> deleteOffsets(
         OffsetDeleteRequestData request,
         BufferSupplier bufferSupplier
     );
+
+    /**
+     * Return the partition index for the given Group.
+     *
+     * @param groupId           The group id.
+     *
+     * @return The partition index.
+     */
+    int partitionFor(String groupId);
+
+    /**
+     * Commit or abort the pending transactional offsets for the given partitions.
+     *
+     * @param producerId        The producer id.
+     * @param partitions        The partitions.
+     * @param transactionResult The result of the transaction.
+     */
+    void onTransactionCompleted(
+        long producerId,
+        Iterable<TopicPartition> partitions,
+        TransactionResult transactionResult
+    );
+
+    /**
+     * Delete the provided partitions' offsets.
```

Comment:
Is it clearer to say "Remove the provided deleted partitions offsets"? 
I guess the parameter explains that it was deleted partitions